### PR TITLE
fix(query): fix values parser in skip_to_next_row

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

 Fix values parser in skip_to_next_row, If `'` is escaped by `\`, it will continue to search a valid row.

Fixes #issue

### CODE REVIEW

File: query/src/sql/statements/value_source.rs [hunk 1]
```diff
@@ -184,8 +184,15 @@ pub fn skip_to_next_row<R: BufferRead>(
             let c = buffer[it];
             reader.consume(it + 1);
 
+            if it == 0 && escaped {
```
Comment (b41sh) [inline query/src/sql/statements/value_source.rs:187]: `(` and `)` can't be escaped.

Comment (sundy-li) [reply query/src/sql/statements/value_source.rs:187]: `escaped` just works for `'`.

Comment (b41sh) [reply query/src/sql/statements/value_source.rs:187]: if the buffer is `\(`, this loop will continue, `balance += 1` will not executed.

**Comment by sundy-li** [comment]
> How can we guarantee it sync?

Oh, I did not notice there is `ValueSourceV2`, maybe we should keep updating v2 until we remove v1 in the future.

**Comment by fkuner** [comment]
> 

Yes, this is awful. Maybe we better remove v1 as soon as possible.

**Pull request merged**